test/foundry/OffchainAssetReceiptVaultFactory.t.sol

```diff
@@ -0,0 +1,57 @@
+pragma solidity 0.8.17;
+
+import "../../contracts/vault/receipt/ReceiptVault.sol";
+import "../../contracts/vault/receipt/ReceiptFactory.sol";
+
+import "forge-std/Test.sol";
+import "forge-std/console.sol";
+
+import "../../contracts/vault/offchainAsset/OffchainAssetReceiptVaultFactory.sol";
+import "../../contracts/vault/offchainAsset/OffchainAssetReceiptVault.sol";
+
+contract OffChainAssetReceiptVaultFactoryTest is Test {
+    OffchainAssetReceiptVaultFactory factory;
+    OffchainAssetVaultConfig offchainAssetVaultConfig;
+    VaultConfig vaultConfig;
+    OffchainAssetReceiptVault implementation;
+    ReceiptFactory receiptFactory;
+    ReceiptVaultFactoryConfig factoryConfig;
+    OffchainAssetReceiptVault vault;
+    address alice;
+
+    function setUp() public {
+        implementation = new OffchainAssetReceiptVault();
+        receiptFactory = new ReceiptFactory();
+
+        // Set up factory config
+        factoryConfig = ReceiptVaultFactoryConfig({
+            implementation: address(implementation),
+            receiptFactory: address(receiptFactory)
+        });
+
+        // Create OffchainAssetReceiptVaultFactory contract
+        factory = new OffchainAssetReceiptVaultFactory(factoryConfig);
+    }
+
+    function testOffchainAssetReceiptVaultFactoryConstuction() public view {
+        assert(address(factory) != address(0));
+    }
+
+    function testCreateChild() public {
+        // Get the first signer address
+        alice = vm.addr(1);
+
+        string memory assetName = "Asset Name";
+        string memory assetSymbol = "ASSET";
+
+        // VaultConfig to create child contract
+        vaultConfig = VaultConfig(address(0), assetName, assetSymbol);
+
+        offchainAssetVaultConfig = OffchainAssetVaultConfig({admin: alice, vaultConfig: vaultConfig});
+
+        vault = factory.createChildTyped(offchainAssetVaultConfig);
```

Comment:
should we be pranking some user? e.g. alice or bob, to check that the admin is who we say it is regardless of who creates the child?